__tests__/pet.test.js

```diff
@@ -0,0 +1,187 @@
+const Pet = require('../src/pet');
+const Parent = require('../src/pet');
+
+describe('constructor',()=>{
+    it('returns an object',()=>{
+        expect(new Pet('Fido')).toBeInstanceOf(Object);
+    })
+   
+
+    it('sets the name property',()=>{
+        const myPet = new Pet('Fido');
+        expect(myPet.name).toEqual('Fido');
+    })
+
+
+    it('has an initial age of 0',()=>{
+        const myPet = new Pet('Fido');
+        expect(myPet.age).toEqual(0);
+    })
+})
+
+describe('growUp',()=>{
+    it('increments the age by 1',()=>{
+        const myPet = new Pet('Fido');
+        myPet.growUp();
+        expect(myPet.age).toEqual(1);
+        expect(myPet.hunger).toEqual(5);
+        expect(myPet.fitness).toEqual(7);
+    })
+})
+
+
+
+describe('fitness',()=>{
+  it('increases fitness by walking by 4',()=>{
+    const myPet=new Pet('Fido');
+    myPet.fitness=4;
+    myPet.walk();
+    expect(myPet.fitness).toEqual(8);
+  })
+
+})
+
+describe('feed',()=>{
+  it('throws an error if my pet is dead',()=>{
+    const myPet = new Pet('Fido');
+    myPet.age=30;
+    expect(()=>myPet.feed()).toThrow('Your pet is no longer alive :(')
+  });
+
+    it('decrease hunger level by 3, minimum hunger is 0',()=>{
+        const myPet = new Pet('Fido');
+        myPet.hunger=9;
+        myPet.age=0;
+        myPet.feed();
+        expect(myPet.hunger).toEqual(6);
+    });
```

Comment:
This test only tests that the hunger level decreases by 3. I would edit this test description to something like `feed decreases hunger level by 3` and then create a separate test to check that the minimum hunger doesn't go below 0, i.e.
```
it('cannot go below hunger of 0', () => {
  ...
});
```